src/bmf.cpp

```diff
@@ -141,48 +145,46 @@ bool CxadbmfPlayer::xadplayer_load()
   }
   else
   {
-    strncpy(bmf.title,xad.title,36);
-    strncpy(bmf.author,xad.author,36);
+    strncpy(bmf.title, xad.title, 36);
+    strncpy(bmf.author, xad.author, 36);
   }
 
   // speed
   if (tune_size - ptr < 1)
     return false;
-  if (bmf.version > BMF0_9B)
-    bmf.speed = tune[ptr++];
-  else
-    bmf.speed = ((tune[ptr++] << 8) / 3) >> 8; // strange, yeh ?
+  bmf.speed = tune[ptr++];
+  if (bmf.version == BMF0_9B)
+    bmf.speed /= 3; // removed pointless shifts from original code
 
   // load instruments
   if (bmf.version > BMF0_9B)
   {
     if (tune_size - ptr < 4)
       return false;
-    unsigned long iflags = (tune[ptr] << 24) | (tune[ptr+1] << 16) | (tune[ptr+2] << 8) | tune[ptr+3];
-    ptr+=4;
+    unsigned long iflags = (tune[ptr] << 24) | (tune[ptr+1] << 16)
+                         | (tune[ptr+2] << 8) | tune[ptr+3];
+    ptr += 4;
 
-    for(i=0;i<32;i++)
+    for (i = 0; i < 32; i++)
+    {
       if (iflags & (1 << (31-i)))
-	  {
+      {
         if (tune_size - ptr < 24)
           return false;
         strncpy(bmf.instruments[i].name, (char *)tune + ptr, 10);
         // 11th char ignored since name must be 0-terminated
         bmf.instruments[i].name[10] = 0;
-        memcpy(bmf.instruments[i].data, &tune[ptr+11], 13);
+        memcpy(bmf.instruments[i].data, tune + ptr + 11, 13);
         ptr += 24;
-	  }
+      }
+      else if (bmf.version == BMF1_1)
+      {
+        memset(bmf.instruments[i].name, 0, 11);
+        memcpy(bmf.instruments[i].data, bmf_default_instrument, 13);
```

Comment:
Is there any way you can replace the hard-coded constants with something more future-proof like `sizeof(bmf_default_instrument)`?

Reply:
Replacing literal constants with `sizeof` (or an `ARRAY_SIZE()` macro, `enum`, `#define`d constant) is always a good idea. But literal numbers are used everywhere, so I didn't bother. I can change it here if you want, but a general literal constant cleanup pass can (and should) be done sometime later. I'd rather fix some real bugs now.

Reply:
Fair enough, I just figured if you were changing those lines anyway it'd be a good opportunity to fix it while you were there.